testsuite/testcases/src/state_sync_performance.rs

```diff
@@ -52,70 +73,154 @@ impl NetworkTest for StateSyncPerformance {
             runtime.block_on(async { fullnode.clear_storage().await })?;
         }
 
-        // Fetch the highest synced version from the swarm
-        let highest_synced_version = runtime.block_on(async {
-            get_highest_synced_version(&ctx.swarm().get_all_nodes_clients_with_names())
-                .await
-                .unwrap_or(0)
-        });
-        if highest_synced_version == 0 {
-            return Err(anyhow::format_err!(
-                "The swarm has synced 0 versions! Something has gone wrong!"
-            ));
-        }
-        info!("Syncing to target version at: {:?}", highest_synced_version);
-
-        // Restart the fullnodes so they start syncing from a fresh state
+        // Restart the nodes so they start syncing from a fresh state
+        // and start the timer.
         for fullnode_id in &all_fullnodes {
             let fullnode = ctx.swarm().full_node_mut(*fullnode_id).unwrap();
             runtime.block_on(async { fullnode.start().await })?;
         }
-
-        // Wait for all fullnodes to catch up to the highest synced version
-        info!("Restarting all the fullnodes and waiting for them to catchup.");
         let timer = Instant::now();
-        runtime.block_on(async {
-            ctx.swarm()
-                .wait_for_all_nodes_to_catchup(fullnode_sync_duration)
-                .await
-        })?;
-        let duration_to_state_sync = timer.elapsed();
-        let seconds_to_state_sync = duration_to_state_sync.as_secs();
 
-        // Calculate the state sync throughput
-        if seconds_to_state_sync == 0 {
+        // Wait for all nodes to catch up to the highest synced version
+        // then calculate and display the throughput results.
+        calculate_and_display_throughput(ctx, timer, self.name())
```

Comment:
nit: this function name is a bit misleading. How about changing it to `ensure_state_sync_throughput` instead?